hazelcast/src/main/java/com/hazelcast/util/ServiceLoader.java

```diff
@@ -36,57 +35,53 @@
 import java.util.List;
 import java.util.Set;
 
+import static com.hazelcast.nio.IOUtil.closeResource;
 import static com.hazelcast.util.Preconditions.isNotNull;
 
 /**
- * Support class for loading Hazelcast services and hooks based on the Java ServiceLoader specification
+ * Support class for loading Hazelcast services and hooks based on the Java {@link ServiceLoader} specification,
  * but changed in the fact of classloaders to test for given services to work in multi classloader
- * environments like application or OSGi servers
+ * environments like application or OSGi servers.
  */
 public final class ServiceLoader {
 
     private static final ILogger LOGGER = Logger.getLogger(ServiceLoader.class);
     private static final String FILTERING_CLASS_LOADER = FilteringClassLoader.class.getCanonicalName();
 
-    // See https://github.com/hazelcast/hazelcast/issues/3922
+    // see https://github.com/hazelcast/hazelcast/issues/3922
     private static final String IGNORED_GLASSFISH_MAGIC_CLASSLOADER =
             "com.sun.enterprise.v3.server.APIClassLoaderServiceImpl$APIClassLoader";
 
     private ServiceLoader() {
     }
 
-    public static <T> T load(Class<T> clazz, String factoryId, ClassLoader classLoader)
-            throws Exception {
-        final Iterator<T> iterator = iterator(clazz, factoryId, classLoader);
+    public static <T> T load(Class<T> clazz, String factoryId, ClassLoader classLoader) throws Exception {
+        Iterator<T> iterator = iterator(clazz, factoryId, classLoader);
         if (iterator.hasNext()) {
             return iterator.next();
         }
         return null;
     }
 
-    public static <T> Iterator<T> iterator(final Class<T> clazz, String factoryId, ClassLoader classLoader)
-            throws Exception {
-        final Set<ServiceDefinition> serviceDefinitions = getServiceDefinitions(factoryId, classLoader);
-
+    public static <T> Iterator<T> iterator(Class<T> clazz, String factoryId, ClassLoader classLoader) throws Exception {
+        Set<ServiceDefinition> serviceDefinitions = getServiceDefinitions(factoryId, classLoader);
         return new NewInstanceIterator<T>(serviceDefinitions, clazz);
     }
 
-    public static <T> Iterator<Class<T>> classIterator(String factoryId, ClassLoader classLoader)
-            throws Exception {
-        final Set<ServiceDefinition> serviceDefinitions = getServiceDefinitions(factoryId, classLoader);
-
+    public static <T> Iterator<Class<T>> classIterator(String factoryId, ClassLoader classLoader) throws Exception {
+        Set<ServiceDefinition> serviceDefinitions = getServiceDefinitions(factoryId, classLoader);
```

Comment:
We definitely need to find a company standard for finals and throws Exception on the next line. Otherwise we'll end up with a never ending pingpong of changes.